### PR TITLE
Restructuring for server rendering.

### PR DESCRIPTION
**Pull request by joeldenning**
This PR is huge (sorry!). It creates a new build for server side rendering that has different methods than the normal build. The new functions are `constructServerLayout` and `renderServerResponseBody`. I've tested them out in a sample project and they seem to be working. There are still more features to build with this, but this provides the foundations.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,12 +33,16 @@
     "test:browser": "cross-env BABEL_ENV=test jest --testEnvironment=jsdom --testPathIgnorePatterns=test/node-only --testPathIgnorePatterns=cypress",
     "test:node": "cross-env BABEL_ENV=test jest --testEnvironment=node --testPathIgnorePatterns=test/browser-only --testPathIgnorePatterns=cypress",
     "test:types": "tsd",
+    "test:exports": "ls-exports path .",
```
Comment (joeldenning) [inline package.json:36]: Cool tool that tests that your package.json is valid - https://github.com/ljharb/list-exports

File: test/node-only/__snapshots__/renderServerResponseBody-node.test.js.snap [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      <footer>
+        <a href="/about">
+          About
+        </a>
+      </footer>
+    </span>
+    <single-spa-router mode="history"
+                       base="/"
+    >
+      <div id="single-spa-application:undefined">
```
Comment (joeldenning) [inline test/node-only/__snapshots__/renderServerResponseBody-node.test.js.snap:36]: The `undefined` is a bug that I'll fix in a future PR

File: test/node-only/__snapshots__/renderServerResponseBody-node.test.js.snap [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    <title>
+      Document
+    </title>
+  </head>
+  <body>
+    <single-spa-router mode="history"
+                       base="/"
+    >
+    </single-spa-router>
+    <script type="systemjs-importmap">
```
Comment (joeldenning) [inline test/node-only/__snapshots__/renderServerResponseBody-node.test.js.snap:75]: `<fragment>` working for import maps (and other things)

File: src/single-spa-layout.js [hunk 4]
```diff
@@ -1,4 +1,3 @@
-export { constructRoutes } from "./constructRoutes.js";
-export { matchRoute } from "./matchRoute.js";
```
Comment (joeldenning) [inline src/single-spa-layout.js:2]: Removed this intentionally - it reduces the bundle size and isn't really needed in the browser.

File: rollup.config.js [hunk 5]
```diff
@@ -1,30 +1,42 @@
...
       globals: {
         "single-spa": "singleSpa",
       },
     },
-    external: ["single-spa"],
+    external: ["single-spa", "path", "fs", "stream", /^parse5.*/],
     plugins: [
-      babel({ babelHelpers: "bundled" }),
+      nodeResolve(),
+      commonjs(),
```
Comment (TheMcMurder) [inline rollup.config.js:38]: Dumb question: Does having these here for the web build change anything with the bundle? AKA Do these need to be conditional based on the `server` flag?

Comment (joeldenning) [reply rollup.config.js:38]: It doesn't change the browser build. Now that I think of it, I'm not sure it's even needed anymore now that I marked everything as webpack externals.

File: src/single-spa-layout.js (hunk 4 above)
Comment (levymateusm) [reply src/single-spa-layout.js:2]: #188 @joeldenning

**Pull request merged**